cfme/physical/physical_server.py

```diff
@@ -0,0 +1,232 @@
+            return self.db_id
+        else:
+            return self.db_id
+
+    def wait_to_appear(self):
+        """Waits for the server to appear in the UI."""
+        view = navigate_to(self.parent, "All")
+        logger.info("Waiting for the server to appear...")
+        wait_for(
+            lambda: self.exists,
+            message="Wait for the server to appear",
+            num_sec=1000,
+            fail_func=view.browser.refresh
+        )
+
+    def wait_for_delete(self):
+        """Waits for the server to remove from the UI."""
+        view = navigate_to(self.parent, "All")
+        logger.info("Waiting for the server to delete...")
+        wait_for(
+            lambda: not self.exists,
+            message="Wait for the server to disappear",
+            num_sec=500,
+            fail_func=view.browser.refresh
+        )
+
+
+@attr.s
+class PhysicalServerCollection(BaseCollection):
+    """Collection object for the :py:class:`cfme.infrastructure.host.PhysicalServer`."""
+
+    ENTITY = PhysicalServer
+
+    def check_physical_servers(self, physical_servers):
+        physical_servers = list(physical_servers)
+        checked_physical_servers = list()
+        view = navigate_to(self, 'All')
+
+        for physical_server in physical_servers:
+            try:
+                view.entities.get_entity(name=physical_server.name, surf_pages=True).check()
+                checked_physical_servers.append(physical_server)
+            except ItemNotFound:
+                raise (ItemNotFound(
+                       'Could not find physical server {} in the UI'.format(physical_server.name)))
+        return view
+
+    def all(self, provider):
+        """returning all physical_servers objects"""
+        view = navigate_to(self, 'All')
+        physical_servers = [self.instantiate(name=item, provider=provider)
+                 for item in view.entities.entity_names]
+        return physical_servers
+
+    def power_on(self, *physical_servers):
+        view = self.check_physical_servers(physical_servers)
+        view.toolbar.power.item_select("Power On", handle_alert=True)
+
+    def power_off(self, *physical_servers):
+        view = self.check_physical_servers(physical_servers)
+        view.toolbar.power.item_select("Power Off", handle_alert=True)
+
+
+@navigator.register(PhysicalServerCollection)
+class All(CFMENavigateStep):
+    VIEW = PhysicalServersView
+    prerequisite = NavigateToAttribute("appliance.server", "LoggedIn")
+
+    def step(self):
+        self.prerequisite_view.navigation.select("Compute", "Physical Infrastructure", "Servers")
+
+
+@navigator.register(PhysicalServer)
+class Details(CFMENavigateStep):
+    VIEW = PhysicalServerDetailsView
+    prerequisite = NavigateToAttribute("parent", "All")
+
+    def step(self):
+        self.prerequisite_view.entities.get_entity(name=self.obj.name, surf_pages=True).click()
+
+
+@navigator.register(PhysicalServer)
+class PolicyAssignment(CFMENavigateStep):
```

Comment:
**Optional**
We don't quite have a consistent definition for this yet, but I'd prefer 'ManagePolicies' for this destination name.

There is some work to be done to PolicyProfileAssignable class, which has a 'ManagePolicies' destination already registered, in order for it to support navigation for classes inheriting it. Its step uses 'All' as a prerequisite, without using `parent` as the target object.

Reply:
@mshriver So I just need to rename this class to 'ManagePolicies'?

Reply:
If you're not too attached to `PolicyAssignment` - easier now while its not in-use anywhere